vllm_spyre/v1/worker/spyre_model_runner.py

```diff
@@ -602,14 +602,11 @@ def _update_states(self, scheduler_output):
 
         # Continuous batching stuff
         for req_id in scheduler_output.finished_req_ids:
-            if req_id in self.req_ids2blocks:
-                logger.debug("Freeing request id: %s", req_id)
-                for freed_block in self.req_ids2blocks[req_id]:
-                    self.free_blocks.append(freed_block)
-                del self.req_ids2blocks[req_id]
-                del self.req_ids2left_pads[req_id]
-
-            del self.requests[req_id]
+            logger.debug("Finishing request id: %s", req_id)
```

Comment:
```suggestion
            if req_id in self.req_ids2blocks:
                logger.debug("Freeing request id: %s", req_id)
```

This debug statement was for specific for CB to have confirmation that the blocks were actually freed. I would suggest to keep it that way. IMO it should not be a general statement that a request has finished, this could probably (or is already) at some other place in the code, not in the CB specific part.